### PR TITLE
Turn on CFStream by default on iOS

### PR DESCRIPTION
**Pull request by muxi**
All projects depending on gRPC will start to use CFStream as transport.

### CODE REVIEW

File: include/grpc/impl/codegen/port_platform.h [hunk 1]
```diff
@@ -229,7 +230,6 @@
 #define GPR_POSIX_SYNC 1
 #define GPR_POSIX_TIME 1
 #define GPR_GETPID_IN_UNISTD_H 1
-/* TODO(mxyan): Remove when CFStream becomes default */
 #ifndef GRPC_CFSTREAM
```
Comment (dapengzhang0) [inline include/grpc/impl/codegen/port_platform.h]: Why TODO is not done but removed?

File: src/objective-c/GRPCClient/GRPCCall.m [hunk 2]
```diff
@@ -434,6 +434,7 @@ + (void)initialize {
   // Guarantees the code in {} block is invoked only once. See ref at:
   // https://developer.apple.com/documentation/objectivec/nsobject/1418639-initialize?language=objc
   if (self == [GRPCCall self]) {
+    setenv(kCFStreamVarName, "1", 1);
```
Comment (dapengzhang0) [inline src/objective-c/GRPCClient/GRPCCall.m]: is `kCFStreamVarName` used/specified anywhere?

Comment (muxi) [reply src/objective-c/GRPCClient/GRPCCall.m]: Yes; it was already in the beginning of this file.

File: include/grpc/impl/codegen/port_platform.h (hunk 1 above)
Comment (muxi) [reply include/grpc/impl/codegen/port_platform.h]: We can only turn CFStream on by default on iOS; the following 3 lines are for non-iOS platforms; they still need this change and we probably will not make CFStream default on them.

**Comment by muxi** [comment]
Basic Tests C++ Linux [dbg] - #17794

**Pull request merged**